packages/vite-plugin-kit-routes/src/lib/plugin.ts

```diff
@@ -609,6 +642,18 @@ export function buildMetadata(
   return baseToReturn
 }
 
+function getSpValue(rawValue: string, param: ExplicitSearchParam) {
+  if ((param.arrayMode ?? 'join') === 'join') {
+    if (param.required || param.default !== undefined) {
+      return `String(${rawValue})`
+    }
+
+    return `${rawValue} !== undefined ? String(${rawValue}) : undefined`
```

Comment:
This could be a function maybe? 'StringOrUndefined'? (so that the generated code is a bit lighter?)

Reply:
Good idea. Implemented!